tests/ui/entry.fixed

```diff
@@ -248,4 +248,11 @@ mod issue14449 {
     }
 }
 
+fn issue15307<K: Eq + Hash, V>(mut m: HashMap<K, V>, k: K, v: V) {
+    if let std::collections::hash_map::Entry::Vacant(e) = m.entry(k) {
+        //~^ map_entry
+        assert!({ e.insert(v); None::<V> }.is_none());
```

Comment:
This is so ugly.

If the user is using the result of `m.insert(…)` which we know to be `None` (as opposed to dropping it), can't we just suggest the same thing as we currently do, but add that `m.insert(…)` would always have returned `None`, and suggest to rewrite the code to not depend on this value or to add an explicit type to `None` if needed?

I dislike suggesting a type from a `rustc_middle::ty::Ty` when it will not be what the user would write when aliases are involved for example.